src/app/ReadClient.h

```diff
@@ -165,6 +165,19 @@ class ReadClient : public Messaging::ExchangeDelegate
          * SendAutoResubscribeRequest is not called, this function will not be called.
          */
         virtual void OnDeallocatePaths(ReadPrepareParams && aReadPrepareParams) {}
+
+        /**
+         * This function is invoked when constructing read/subscribeRequest, where the ReadClient would use cached cluster data
+         * version to construct data version filter list. When the particular cached cluster data version intersects with user
```

Comment:
Had further discussions with @yunhanw-google after this. We came to the following consensus:

- Applications have the option of either specifying explicitly a data version filter list in the prepare params OR, set that to null and trigger this alternative 'callback' pathway to retrieve them. There is no need to do this extra special merge logic in this PR since I don't think most applications would use it that in a mixed modality.

- We'll pass in the requested paths through this method to limit the set of data version filters the cache will pick-up from its store. This ensures that a cache that is used across different reads/subscriptions with different paths in them will not result in the cache picking up the wrong filters to send that don't actually intersect with the specified paths in the request. The alternative was to more tightly couple the lifetime of the AttributeCache with a given ReadClient instance, but that we agreed is too limiting, and difficult to enforce.

Reply:
Done, also add test as well, thanks